### PR TITLE
feat: adds A/B testing to develop

### PR DESCRIPTION
**Pull request by scefali**

### CODE REVIEW

File: src/docs/ab-testing/index.mdx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+You can then add checks on the value of `exposed` to drive your in-app logic.
+
+### Logging the Experiment on the Backend
+
+Logging the experiment on the backend can be tricky because it creates an analytics event each time it's triggered. That might be too many events depending on how often it's called. Refer to the implementation of the [log experiment endpoint](https://github.com/getsentry/getsentry/blob/master/getsentry/web/experiment.py) to see how to log the experiment on the backend.
+
+### Alternatives to Logging the Experiment
+
+If the volume of analytics events is too high, we can track experiments indirectly via other methods. For example, if your test impacts the
```
Comment (ceorourke) [inline src/docs/ab-testing/index.mdx:135]: Unless you know what this was meant to say, we might as well leave it out.

File: src/docs/ab-testing/index.mdx [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    def assign(self, params, org, actor):
+        if settings.IS_TEST or org.slug == "sentry":
+            params.variant = "1"
+        else:
+            params.variant = WeightedChoice(
+                choices=["0", "1"], weights=[1 - WEIGHT_VARIANT, WEIGHT_VARIANT], unit=org.id
+            )
+```
+
+A few things
```
Comment (ceorourke) [inline src/docs/ab-testing/index.mdx:67]: Probably leave this out too (just this line)

File: src/docs/ab-testing/index.mdx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+- Assignment logic: The `assign` function on the experiment class determines which group an entity falls into. In the most simple case, you have just two groups: a control and test group. If that's the case, you can use `BernoulliTrial` to generate the test value. If you have three or more groups, you can use `UniformChoice`. You can also insert whatever custom logic you want, such as enabling an experiment if a particular feature flag is set.
+- You can use other random assignment operators. Learn more: [https://facebook.github.io/planout/docs/random-operators.html](https://facebook.github.io/planout/docs/random-operators.html)
+
+We are setting the values to "0" or "1" and giving a 50% chance an organization sees the plugins in the Integration Directory
+
+Initialize the experiment in `getsentry/web/apps.py` :
+
+```python
+class Config(AppConfig):
+		.
```
Comment (ceorourke) [inline src/docs/ab-testing/index.mdx:79]: This looks kinda strange on the page imo, do you think this was meant to be `...`? 

Screenshot 2023-05-04 at 3 42 50 PM

File: src/docs/ab-testing/index.mdx [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+## Example
+
+- Adding a "signup with github” button to the sign up page
+- Hypothesis: signup rate will increase because there's an easier way to sign up
+
+We showed half the visitors the new signup page, and the other half the old page. Turns out there was no effect on the signup rate, i.e. people do use GitHub to sign up, but if it isn't available they will use the regular form instead.
+
+## Details
+
+Our A/B testing framework is built on [Facebook's planout library](http://facebook.github.io/planout/).
```
Comment (roggenkemper) [inline src/docs/ab-testing/index.mdx:19]: this link just leads to a meta website about open source right now

File: src/docs/ab-testing/index.mdx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+To launch the A/B test you will need to do the following:
+
+1. Set up an A/B test in Getsentry to place users/organizations into different test groups.
+2. Adding the experiment to the list of experiments in Sentry for Typescript definitions
+3. Add the logic where you have to control the in-app behavior
+
+### Changes to `getsentry/getsentry`
+
+In **getsentry/getsentry**, locate the  `getsentry/experiments/config.py` file). In this file, you will define an experiment class that inherits from either `OrgExperiment` or `UserExperiment`. Choose one depending on whether you want to assign the experiment across each user or each organization.
```
Comment (roggenkemper) [inline src/docs/ab-testing/index.mdx:38]: extra `)`

**Review by roggenkemper — APPROVED** [review]

**Pull request merged**